### PR TITLE
Implement option to disable sorting on ChoiceSet

### PR DESCRIPTION
**Pull request by LebombJames**
Currently, ChoiceSets always sort choices, which is fine most of the time, but sometimes you want an arbitrary order. A good example of this is Oracle Curses - it would make more sense if they were sorted Minor -> Extreme.

This merge adds a `"sort"` boolean to the ChoiceSet RE, which defaults to true. When set to false, the prompt will follow the order declared in the ChoiceSet RE itself.

Left is `"sort": true` (or omitted), right is `"sort": false`

image

(First real code merge, hope I haven't broken anything 🙏)

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Personal opinion is that explicit choice sets should not be sorted, and queried ones should be. But this is shark's baby, so up to him to decide.

File: src/module/rules/rule-element/choice-set/rule-element.ts [hunk 1]
```diff
@@ -42,6 +45,7 @@ class ChoiceSetRuleElement extends RuleElementPF2e {
         this.allowedDrops = isObject(data.allowedDrops) ? new PredicatePF2e(data.allowedDrops) : new PredicatePF2e();
         this.allowNoSelection = !!data.allowNoSelection;
         this.rollOption = typeof data.rollOption === "string" && data.rollOption ? data.rollOption : null;
+        this.sort = !!(data.sort ?? true);
```
Comment (stwlam) [inline src/module/rules/rule-element/choice-set/rule-element.ts:48]:
```suggestion
        this.sort = !!data.sort;
```

**Comment by stwlam** [comment]
> Personal opinion is that explicit choice sets should not be sorted, and queried ones should be, rather than introducing a sort parameter. But this is shark's baby, so up to him to decide.

Yeah, I think that'd make for the best results.

**Comment by CarlosFdez** [comment]
#3678 was chosen instead, using the given arbitrary order whenever there is a choice list. While this should solve your issue, thank you for making this PR and bringing up the issue.

**Pull request closed without merging**